hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -198,8 +206,22 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
 
       setHmsTableParameters(newMetadataLocation, tbl, metadata.properties(), removedProps, hiveEngineEnabled);
 
-      persistTable(tbl, updateHiveTable);
-      threw = false;
+      try {
+        persistTable(tbl, updateHiveTable);
+        commitStatus = CommitStatus.SUCCESS;
+      } catch (Throwable persistFailure) {
+        LOG.error("Cannot tell if commit succeeded, attempting to reconnect and check", persistFailure);
+        commitStatus = checkCommitStatus(newMetadataLocation);
```

Comment:
There's retries built into the `HiveClientPool` ,  thoughts on why an additional retry helps, @aokolnychyi , @RussellSpitzer ?

Reply:
This was from the discussion from the issue with @marton-bod and @pvary, the thought was we should give it one more chance just incase a full new connection attempt would succeed because the cost of failure is that the user needs to do manual investigation, this design also lets us slot in other behaviors later if we decide that makes sense.